File: framework/web/src/test/java/cn/bestwu/simpleframework/support/packagescan/PackageScanClassResolverTest.java
package cn.bestwu.simpleframework.support.packagescan;

import cn.bestwu.simpleframework.web.Response;
import org.junit.Test;
import org.springframework.core.io.Resource;
import org.springframework.core.io.support.PathMatchingResourcePatternResolver;

/**
 * @author Peter Wu
 */
public class PackageScanClassResolverTest {

  PackageScanClassResolver packageScanClassResolver = new PackageScanClassResolver();

  @Test
  public void findClass() {
    System.err.println(packageScanClassResolver.findImplementations(Response.class, "cn.bestwu"));
  }

  @Test
  public void findResource() throws Exception {
    String target = "*/**/*";
    Resource[] resources = new PathMatchingResourcePatternResolver(
        Response.class.getClassLoader())
        .getResources("classpath*:" + target + ".class");
    for (Resource resource : resources) {
      System.err.println(resource.getURI());
    }
  }

  @Test
  public void findClass2() {

  }
}

File: web/src/test/kotlin/top/bettercode/summer/test/web/support/packagescan/PackageScanClassResolverTest.kt
package top.bettercode.summer.test.web.support.packagescan

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.core.io.support.PathMatchingResourcePatternResolver
import top.bettercode.summer.web.Response
import top.bettercode.summer.web.support.packagescan.PackageScanClassResolver

/**
 * @author Peter Wu
 */
class PackageScanClassResolverTest {
    val packageScanClassResolver = PackageScanClassResolver()

    @Test
    fun findClass() {
        val findImplementations =
            packageScanClassResolver.findImplementations(Response::class.java, "top.bettercode")
        Assertions.assertTrue(findImplementations.isNotEmpty())
//        System.err.println(findImplementations)
    }

    @Test
    fun findResource() {
        val target = "*/**/*"
        val resources = PathMatchingResourcePatternResolver(
            Response::class.java.classLoader
        ) .getResources("classpath*:$target.class")
        Assertions.assertTrue(resources.isNotEmpty())
//        for (resource in resources) {
//            try {
//                System.err.println(resource.uri)
//            } catch (_: Exception) {
//            }
//        }
    }

}